jobs/joint_genotyping.py

```diff
@@ -0,0 +1,735 @@
+    job_attrs: dict | None = None,
+) -> tuple[Job, set[str]]:
+    """
+    Add GVCFs to a genomics database (or create a new instance if it doesn't exist)
+    Returns a Job, or None if no new samples to add
+    """
+    rm_cmd = ''
+
+    if updating_existing_db and not overwrite:
+        # Update existing DB
+        genomicsdb_param = f'--genomicsdb-update-workspace-path {genomicsdb_gcs_path}'
+        job_name = 'Joint genotyping: adding to GenomicsDB'
+        msg = f'Adding {len(sample_names_to_add)} samples: {", ".join(sample_names_to_add)}'
+        if sample_names_to_skip:
+            msg += (
+                f'Skipping adding {len(sample_names_to_skip)} samples that are already '
+                f'in the DB: {", ".join(sample_names_to_skip)}"'
+            )
+    else:
+        # Initiate new DB
+        job_name = 'Joint genotyping: creating GenomicsDB'
+
+        genomicsdb_param = f'--genomicsdb-workspace-path {genomicsdb_gcs_path}'
+        # Need to remove the existing database if exists
+        rm_cmd = (
+            f'gsutil ls {genomicsdb_gcs_path} && gsutil -q rm -rf {genomicsdb_gcs_path}'
+        )
+        msg = (
+            f'Creating a new DB with {len(sample_names_to_add)} samples: '
+            f'{", ".join(sample_names_to_add)}'
+        )
+
+    sample_map = b.read_input(str(sample_map_bucket_path))
+
+    j = b.new_job(job_name, job_attrs)
+    j.image(image_path('gatk'))
+
+    if depends_on:
+        j.depends_on(*depends_on)
+
+    # The Broad: testing has shown that the multithreaded reader initialization
+    # does not scale well beyond 5 threads, so don't increase beyond that.
+    nthreads = 5
+
+    # The Broad: The memory setting here is very important and must be several
+    # GiB lower than the total memory allocated to the VM because this tool uses
+    # a significant amount of non-heap memory for native libraries.
+    xms_gb = 8
+    xmx_gb = 25
+
+    STANDARD.set_resources(j, nthreads=nthreads, mem_gb=xmx_gb + 1)
+
+    params = [
+        # The Broad: We've seen some GenomicsDB performance regressions related
+        # to intervals, so we're going to pretend we only have a single interval
+        # using the --merge-input-intervals arg. There's no data in between since we
+        # didn't run HaplotypeCaller over those loci, so we're not wasting any compute
+        '--merge-input-intervals',
+        '--consolidate',
+        # The batch_size value was carefully chosen here as it is the optimal value for
+        # the amount of memory allocated within the task; please do not change it
+        # without consulting the Hellbender (GATK engine) team!
+        '--batch-size',
+        '50',
+        # Using cloud + --consolidate causes a TileDB error:
+        # https://github.com/broadinstitute/gatk/issues/7653
+        # However disabling --consolidate decreases performance of reading, cause
+        # GenotypeGVCFs to run forever. So instead, using non-cloud GenomicsDB.
```

Comment:
Did you mean to put this comment above `--consolidate` or maybe have a commented-out `--cloud` option?

You also mention this already in line 100, should that comment be removed?

Reply:
I can actually just remove all the code for interacting with the GenomicsDB on cloud, we are not using it anyway. I'll move this comment into the docstring so we don't forget why we are not using the cloud mode.